### PR TITLE
feat(api-request-builder): support version when deleting

### LINKED ISSUE #112: [api-request-builder] Support version when deleting
### Description

Currently to do a delete, you have to manually add the version to the uri.

I find the user experience of this very bad, especially when compared to the old SDK

### Expected Behavior

Should expose a method like service.delete(versionNumber).build()
### Current Behavior

Example of current behaviour can be found [here](https://github.com/commercetools/nodejs/blob/6dce24c25665289ae7d21329a6f1b026b5a7096f/integration-tests/sdk/channels.it.js#L141)

### PR DESCRIPTION
**Pull request by wizzy25**
#### Description

BREAKING CHANGE:
Expose a method `.toDelete()` to pass in the version of an item to delete instead of passing it
directly to the uri

#### Todo
- Tests
    - [x] Unit
    - [x] Integration
- [x] Documentation

resolves #112

### CODE REVIEW

**Comment by junajan** [comment]
Hello @wizzy25 .. am I correct that the method is called `toDelete` instead of just `delete` because the second one is a reserved word in JS?

**Comment by wizzy25** [comment]
@junajan it's called `.toDelete()`

**Comment by emmenko** [comment]
@wizzy25 can you explain the motivation for that?

PS: next time it would be nice to create an issue first, so that it can be discussed, before working on the PR. Thanks ;)

File: packages/api-request-builder/src/delete.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * in order to generate a delete uri
+ *
+ * @param  {number} version - The version of the item
+ * @throws if `version` is missing or not a number.
+ * @return {Object} The instance of the service, can be chained.
+ */
+
+export default function toDelete (version: number): Object {
+  if (!version || typeof version !== 'number')
+    throw new Error('Required version to delete missing or not valid')
```
Comment (junajan) [inline packages/api-request-builder/src/delete.js]: This error message looks strange to me: `Required version to delete missing or not valid`
What about to use something like this: `A resource version is invalid or missing`

But let's wait for others what do they think about it.

**Comment by wizzy25** [comment]
@emmenko The [issue](https://github.com/commercetools/nodejs/issues/112) is 19 days old :-D

File: packages/api-request-builder/src/delete.js (hunk 1 above)
Comment (wizzy25) [inline packages/api-request-builder/src/delete.js]: @junajan I think the version is [only?] needed in the uri with a `DELETE` request

Comment (junajan) [inline packages/api-request-builder/src/delete.js]: Yep, but I would expect there a word "IS" like this: `Required version to delete IS missing or it is not valid`

Comment (wizzy25) [inline packages/api-request-builder/src/delete.js]: semantics................ @junajan :')

Comment (junajan) [inline packages/api-request-builder/src/delete.js]: oki :grimacing:

**Comment by emmenko** [comment]
@wizzy25 totally missed the link, sorry 😅

File: integration-tests/cli/csv-parser-price.it.js [hunk 2]
```diff
@@ -55,8 +55,7 @@ describe('CSV and CLI Tests', () => {
       })
     })
 
-    // FIXME: fails with `ERR! No type with key \'custom-type\' found`
-    test.skip('should take input from file', (done) => {
+    test('should take input from file', (done) => {
```
Comment (emmenko) [inline integration-tests/cli/csv-parser-price.it.js]: Why is this change here?

File: packages/api-request-builder/src/delete.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Set the `version` number to the internal state of the service instance
+ * in order to generate a delete uri
+ *
+ * @param  {number} version - The version of the item
+ * @throws if `version` is missing or not a number.
+ * @return {Object} The instance of the service, can be chained.
+ */
+
+export default function toDelete (version: number): Object {
```
Comment (emmenko) [inline packages/api-request-builder/src/delete.js]: Why not just name this function `version`? At the end it's exactly what it does.

```js
service.byId('123').version(1).build()

// or
service.byId('123').withVersion(1).build()
```

File: packages/api-request-builder/src/delete.js (hunk 1 above)
Comment (emmenko) [inline packages/api-request-builder/src/delete.js]: I wouldn't mention any "delete" thingy in this function. If we are considering `version` as just a query parameter, it can be used for anything, not just "delete".

So I would keep this generic. See also my other comment above.

File: docs/sdk/README.md [hunk 4]
```diff
@@ -146,9 +146,13 @@ describe('Channels', () => {
   })
 
   it('delete', () => {
-    const uri = service.byId(channelResponse.id).build()
+    const uri = service
+      .byId(channelResponse.id)
+      .toDelete(channelResponse.version)
```
Comment (hisabimbola) [inline docs/sdk/README.md]: Let's go with `delete()` as least to maintain a similar API with the old SDK...

File: packages/api-request-builder/src/delete.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  if (!version || typeof version !== 'number')
```
Comment (hisabimbola) [inline packages/api-request-builder/src/delete.js]: You can just do

```
if (typeof version !== 'number)
```

achieves same...

File: packages/api-request-builder/src/delete.js (hunk 1 above)
Comment (hisabimbola) [inline packages/api-request-builder/src/delete.js]: Let's use this

`A resource version is invalid or missing` or something like this. This explains more what *version* is and easily connected to the platform definition of version

File: docs/sdk/README.md (hunk 4 above)
Comment (hisabimbola) [inline docs/sdk/README.md]: Scratch this, I go with @emmenko suggestion...

File: integration-tests/cli/csv-parser-price.it.js (hunk 2 above)
Comment (wizzy25) [inline integration-tests/cli/csv-parser-price.it.js]: I just fixed all tests in this PR... Okay then, I'd create a separete PR for this 👍🏾

File: docs/sdk/README.md (hunk 4 above)
Comment (adnasa) [inline docs/sdk/README.md]: @hisabimbola shouldn't you be in vacation? :)

**Comment by wizzy25** [comment]
Implemented changes... Kindly review again 
cc @junajan @emmenko @adnasa @hisabimbola(Enjoy your holiday;))

**Comment by emmenko** [comment]
🤔 did you push? there is no new commit...

**Comment by wizzy25** [comment]
😃😃 Pushed now @emmenko
